### PR TITLE
[SPARK-34402][SQL] Group exception about data format schema

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Group exception about data format schema of different format, orc/parquet

### Why are the changes needed?
group exception

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Not need

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @cloud-fan And will restart https://github.com/apache/spark/pull/30869

**Comment by AngersZhuuuu** [comment]
FYI @cloud-fan

**Comment by AngersZhuuuu** [comment]
retest this please

**Comment by AngersZhuuuu** [comment]
ping @maropu @cloud-fan @beliefer

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala [hunk 1]
```diff
@@ -586,11 +586,10 @@ private[sql] object ParquetSchemaConverter {
 
   def checkFieldName(name: String): Unit = {
     // ,;{}()\n\t= and space are special characters in Parquet schema
-    checkConversionRequirement(
-      !name.matches(".*[ ,;{}()\n\t=].*"),
-      s"""Attribute name "$name" contains invalid character(s) among " ,;{}()\\n\\t=".
```
Comment (beliefer) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala:591]: `Attribute` -> `Column` ?

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala [hunk 2]
```diff
@@ -3002,8 +3002,8 @@ class HiveDDLSuite
           spark.sql(s"INSERT OVERWRITE LOCAL DIRECTORY '${path.getCanonicalPath}' " +
             s"STORED AS PARQUET SELECT ID, if(1=1, 1, 0), abs(id), '^-' FROM v")
         }.getMessage
-        assert(e.contains("Attribute name \"(IF((1 = 1), 1, 0))\" contains" +
-          " invalid character(s) among \" ,;{}()\\n\\t=\". Please use alias to rename it."))
+        assert(e.contains("Column name \"(IF((1 = 1), 1, 0))\" contains invalid character(s) " +
```
Comment (beliefer) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala:3005]: """Column name "(IF((1 = 1), 1, 0))"..." ,;{}()\\n\\t="..."""

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala (hunk 1 above)
Comment (AngersZhuuuu) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala:591]: Not under stand what you mean ==

Comment (beliefer) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala:591]: Why you change the message?

Comment (AngersZhuuuu) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala:591]: > Why you change the message?

unify such schema column name error and make same error message

Comment (beliefer) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala:591]: So this change makes break change? Maybe we need update migrate docs.

**Comment by AngersZhuuuu** [comment]
retest this please

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala [hunk 3]
```diff
@@ -3002,8 +3002,8 @@ class HiveDDLSuite
...
+          "among \" ,;{}()\\n\\t=\"\". Please use alias to rename it."))
```
Comment (cloud-fan) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala:3006]: It looks a bit weird to see `among " ,;{}()\\n\\t="`. Can we exclude it from the error message?

Comment (AngersZhuuuu) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala]: > exclude

Done

**Review by cloud-fan — APPROVED** [review]

**Comment by AngersZhuuuu** [comment]
retest this please

**Comment by AngersZhuuuu** [comment]
ping @cloud-fan

**Comment by AngersZhuuuu** [comment]
Any more suggestion?

**Comment by cloud-fan** [comment]
thanks, merging to master/3.2!

**Pull request closed without merging**